src/context/cart-context.js

```diff
@@ -0,0 +1,87 @@
+import {createContext, useContext, useReducer} from "react";
+import { cartReducer } from "../reducer";
+import axios from "axios";
+
+const CartContext = createContext(null);
+
+const encodedToken = localStorage.getItem("encodedToken");
```

Comment:
Instead of fetching token from local storage everytime you can access it from auth

Reply:
I haven't added authentication yet. I will store it in auth context.